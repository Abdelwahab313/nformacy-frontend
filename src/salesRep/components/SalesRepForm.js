import Container from '@material-ui/core/Container';
import CssBaseline from '@material-ui/core/CssBaseline';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import React from 'react';
import useForm from 'react-hook-form';
import useSalesRepFormStyle from '../styles/salesRepForm';

const SalesRepForm = ({
  pageTitle,
  onSubmit,
  formType,
  first_name,
  last_name,
  phone_number,
  national_id,
  username,
}) => {
  const { register, handleSubmit, errors, getValues } = useForm();
  const classes = useSalesRepFormStyle();

  return (
    <Container component='main' maxWidth='xs' dir={'rtl'}>
      <CssBaseline />
      <div className={classes.paper}>
        <Typography component='h1' variant='h5'>
          {pageTitle}
        </Typography>
        <form
          onSubmit={handleSubmit(onSubmit)}
          id={'addUserForm'}
          className={classes.form}
          noValidate>
          <Grid container spacing={2}>
            <Grid
              container
              direction='row'
              spacing={2}
              className={classes.nameGrid}>
              <Grid item xs={6}>
                <TextField
                  inputProps={{ tabIndex: '1', 'data-testid': 'first_name' }}
                  variant='outlined'
                  fullWidth
                  id='firstName'
                  label='الاسم الاول'
                  name='first_name'
                  defaultValue={first_name}
                  autoComplete='fname'
                  inputRef={register({
                    required: 'برجاء ادخال اسم الموظف الأول',
                  })}
                  error={!!errors.first_name}
                  helperText={errors.first_name && errors.first_name.message}
                  autoFocus
                />
              </Grid>
              <Grid item xs={6}>
                <TextField
                  inputProps={{ tabIndex: '2', 'data-testid': 'last_name' }}
                  autoComplete='lname'
                  name='last_name'
                  defaultValue={last_name}
                  variant='outlined'
                  fullWidth
                  id='lastName'
                  label='الاسم الاخير'
                  inputRef={register({
                    required: 'برجاء ادخال اسم الموظف الاخير',
                  })}
                  error={!!errors.last_name}
                  helperText={errors.last_name && errors.last_name.message}
                />
              </Grid>
            </Grid>
            {formType === 'ADD' && (
              <Grid container item spacing={2}>
                <Grid item xs={12}>
                  <TextField
                    inputProps={{ tabIndex: '3', 'data-testid': 'password' }}
                    variant='outlined'
                    fullWidth
                    name='password'
                    label='كلمه المرور'
                    type='password'
                    id='password'
                    autoComplete='current-password'
                    inputRef={register({ required: 'برجاء ادخال كلمه السر' })}
                    error={!!errors.password}
                    helperText={errors.password && errors.password.message}
                  />
                </Grid>
                <Grid item xs={12}>
                  <TextField
                    inputProps={{
                      tabIndex: '3',
                      'data-testid': 'confirm-password',
                    }}
                    variant='outlined'
                    fullWidth
                    name='confirmPassword'
                    label='تاكيد كلمة المرور'
                    type='password'
                    id='confirm-password'
                    inputRef={register({
                      required: 'برجاء تاكيد ادخال كلمة المرور',
                      validate: (value) =>
                        value === getValues().password ||
                        'برجاء تاكد من تطابق كلمه المرور',
                    })}
                    error={!!errors.confirmPassword}
                    helperText={
                      errors.confirmPassword && errors.confirmPassword.message
                    }
                  />
                </Grid>
              </Grid>
            )}
            <Grid item xs={12}>
              <TextField
                inputProps={{ tabIndex: '4', 'data-testid': 'national_id' }}
                variant='outlined'
                fullWidth
                name='national_id'
                defaultValue={national_id}
                label='الرقم القومى'
                id='nationalId'
                inputRef={register({
                  required: 'برجاء ادخال الرقم القومى',
                  pattern: {
                    value: /^[0-9]{14}$/,
                    message: 'الرقم يجب ان يكون ١٤ رقم',
                  },
                })}
                error={!!errors.national_id}
                helperText={errors.national_id && errors.national_id.message}
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                inputProps={{ tabIndex: '5', 'data-testid': 'username' }}
                variant='outlined'
                fullWidth
                name='username'
                defaultValue={username}
                label='اسم تسجيل الدخول'
                id='userName'
                inputRef={register({
                  required: 'برجاء ادخال اسم تسجيل الدخول',
                  pattern: {
                    value: /^[^0-9][a-zA-Z0-9_]+$/,
                    message: 'اسم تسجيل الدخول غير صالح',
                  },
                })}
                error={!!errors.username}
                helperText={errors.username && errors.username.message}
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                inputProps={{ tabIndex: '6', 'data-testid': 'phone_number' }}
                variant='outlined'
                fullWidth
                name='phone_number'
                defaultValue={phone_number}
                label='رقم الموبايل'
                id='phone_number'
                inputRef={register({
                  required: 'برجاء ادخال رقم الهاتف',
                  pattern: {
                    value: /(01)(?=\d{9}$)\d+$/,
                    message: 'رقم الهاتف غير صحيح',
                  },
                })}
                error={!!errors.phone_number}
                helperText={errors.phone_number && errors.phone_number.message}
              />
            </Grid>
          </Grid>
          <Button
            tabIndex='7'
            id={'save_sales_rep_submit_btn'}
            type='submit'
            fullWidth
            variant='contained'
            color='primary'
            className={classes.submit}>
            حفظ
          </Button>
        </form>
      </div>
    </Container>
  );
};

export default SalesRepForm;
