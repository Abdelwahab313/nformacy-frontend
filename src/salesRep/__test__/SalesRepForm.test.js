import SalesRepForm from '../components/SalesRepForm';
import React from 'react';
import { render } from '@testing-library/react';

describe('SalesRep Form', () => {
  beforeEach(() => {
    return require('mutationobserver-shim');
  });
  it('should render form with no data if props not passed to form', () => {
    const onSubmit = jest.fn();
    let pageTitle = 'اضافه موظف جديد';
    const { asFragment } = render(
      <SalesRepForm
        pageTitle={pageTitle}
        onSubmit={onSubmit}
        formType={'ADD'}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
