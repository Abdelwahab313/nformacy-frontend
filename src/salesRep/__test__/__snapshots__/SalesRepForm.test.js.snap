// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SalesRep Form should render form with no data if props not passed to form 1`] = `
<DocumentFragment>
  <main
    class="MuiContainer-root MuiContainer-maxWidthXs"
    dir="rtl"
  >
    <div
      class="makeStyles-paper-1"
    >
      <h1
        class="MuiTypography-root MuiTypography-h5"
      >
        اضافه موظف جديد
      </h1>
      <form
        class="makeStyles-form-4"
        id="addUserForm"
        novalidate=""
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root makeStyles-nameGrid-2 MuiGrid-container MuiGrid-spacing-xs-2"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused"
                  data-shrink="true"
                  for="firstName"
                  id="firstName-label"
                >
                  الاسم الاول
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="fname"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    data-testid="first_name"
                    id="firstName"
                    name="first_name"
                    tabindex="1"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-207 MuiOutlinedInput-notchedOutline"
                    style="padding-left: 8px;"
                  >
                    <legend
                      class="PrivateNotchedOutline-legend-208"
                      style="width: 0px;"
                    >
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink="false"
                  for="lastName"
                  id="lastName-label"
                >
                  الاسم الاخير
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="lname"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    data-testid="last_name"
                    id="lastName"
                    name="last_name"
                    tabindex="2"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-207 MuiOutlinedInput-notchedOutline"
                    style="padding-left: 8px;"
                  >
                    <legend
                      class="PrivateNotchedOutline-legend-208"
                      style="width: 0.01px;"
                    >
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink="false"
                  for="password"
                  id="password-label"
                >
                  كلمه المرور
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="current-password"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    data-testid="password"
                    id="password"
                    name="password"
                    tabindex="3"
                    type="password"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-207 MuiOutlinedInput-notchedOutline"
                    style="padding-left: 8px;"
                  >
                    <legend
                      class="PrivateNotchedOutline-legend-208"
                      style="width: 0.01px;"
                    >
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink="false"
                  for="confirm-password"
                  id="confirm-password-label"
                >
                  تاكيد كلمة المرور
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    data-testid="confirm-password"
                    id="confirm-password"
                    name="confirmPassword"
                    tabindex="3"
                    type="password"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-207 MuiOutlinedInput-notchedOutline"
                    style="padding-left: 8px;"
                  >
                    <legend
                      class="PrivateNotchedOutline-legend-208"
                      style="width: 0.01px;"
                    >
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
                for="nationalId"
                id="nationalId-label"
              >
                الرقم القومى
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="national_id"
                  id="nationalId"
                  name="national_id"
                  tabindex="4"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-207 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-208"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
                for="userName"
                id="userName-label"
              >
                اسم تسجيل الدخول
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="username"
                  id="userName"
                  name="username"
                  tabindex="5"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-207 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-208"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
                for="phone_number"
                id="phone_number-label"
              >
                رقم الموبايل
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="phone_number"
                  id="phone_number"
                  name="phone_number"
                  tabindex="6"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-207 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-208"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-5 MuiButton-containedPrimary MuiButton-fullWidth"
          id="save_sales_rep_submit_btn"
          tabindex="7"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            حفظ
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </main>
</DocumentFragment>
`;
